service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/ControlledAzureDatabaseResource.java

```diff
@@ -264,6 +300,11 @@ public void validate() {
       throw new MissingRequiredFieldException(
           "Missing required databaseOwner field for shared ControlledAzureDatabase.");
     }
+    if (getAccessScope() == AccessScopeType.ACCESS_SCOPE_PRIVATE
+        && getAllowAccessForAllWorkspaceUsers()) {
```

Comment:
I don't understand how `allowAccessForAllWorkspaceUsers` is meant to work-- as far as I can tell, this is the only place it's every checked. So what difference does setting it to be true or false make?

Reply:
the feature that will use `allowAccessForAllWorkspaceUsers` has yet to be implemented. Some databases (the cromwell use case) should be accessible to other workspace users. Other databases (the WDS use case) should only be accessible to the database owner identity.

applicable text from the design doc:

> A database can be created as a private resource or a shared resource. As a private resource, the user’s pet is the owner. As a shared resource, a Workspace Identity is the owner and additionally the database may be configured to allow access by other identities.

Reply:
Ah, I missed the iterative part of the introduction of `allowAccessForAllWorkspaceUsers`.